src/store/actions/transactions.js

```diff
@@ -56,28 +56,24 @@ export const transactionsRetrieved = ({
     offset,
   };
 
-  getTransactions({ network, params }, token)
-    .then((response) => {
-      dispatch({
-        type: actionTypes.transactionsRetrieved,
-        data: {
-          offset,
-          address,
-          filters,
-          confirmed: response.data,
-          count: response.meta.total,
-        },
-      });
-    })
-    .catch((error) => {
-      dispatch({
-        type: actionTypes.transactionLoadFailed,
-        data: { error },
-      });
-    })
-    .finally(() => {
-      dispatch(loadingFinished(actionTypes.transactionsRetrieved));
+  try {
+    const { data, meta } = await getTransactions({ network, params }, token);
+    dispatch({
+      type: actionTypes.transactionsRetrieved,
+      data: {
+        offset,
+        address,
+        filters,
+        confirmed: data,
+        count: meta.total,
+      },
+    });
+  } catch (error) {
+    dispatch({
+      type: actionTypes.transactionLoadFailed,
+      data: { error },
     });
+  }
```

Comment:
add finally here and keep the `dispatch(loadingFinished `

Reply:
Done